metadata-db-app/src/migrations/057_regenerate_collection_identifiers.clj

```diff
@@ -44,21 +46,17 @@
                        (util/gzip-blob->string)
                        (edn/read-string))
           {{:keys [collection-identifier collection-applicable granule-applicable provider-id]} :catalog-item-identity} metadata
-          concept-ids (if (or collection-applicable granule-applicable)
-                        (get-concept-ids provider-id collection-identifier)
-                        nil)
-          entry-titles (if (or collection-applicable granule-applicable)
-                        (get-entry-titles provider-id collection-identifier)
-                        nil)
+          collections-by-entry-titles (when (or collection-applicable granule-applicable)
+                                        (get-concept-ids provider-id collection-identifier))
+          collections-by-concept-ids (when (or collection-applicable granule-applicable)
+                                       (get-entry-titles provider-id collection-identifier))
+          collections (distinct (remove nil? (concat collections-by-entry-titles collections-by-concept-ids)))
           metadata (if deleted
                      metadata
-                     (if (seq concept-ids)
-                       (assoc-in metadata [:catalog-item-identity :collection-identifier :concept-ids] concept-ids)
-                       metadata))
-          metadata (if deleted
-                     metadata
-                     (if (seq entry-titles)
-                       (assoc-in metadata [:catalog-item-identity :collection-identifier :entry-titles] entry-titles)
+                     (if (seq collections)
+                       (-> metadata
+                           (assoc-in [:catalog-item-identity :collection-identifier :concept-ids] (map :concept_id collections))
```

Comment:
It looks like this is to fix cases where you had some entry-titles and some concept-ids in an ACL, but not a full list of both?